domain/include/cstone/sfc/box.hpp

```diff
@@ -101,23 +104,33 @@ HOST_DEVICE_FUN constexpr int pbcDistance(int x)
 template<class T>
 class Box
 {
+    HOST_DEVICE_FUN void validateBoundaries(int boundaryX, int boundaryY, int boundaryZ)
+    {
+        if (boundaryX < 0 || boundaryX > 2 || boundaryY < 0 || boundaryY > 2 || boundaryZ < 0 || boundaryZ > 2)
+        {
+            throw std::runtime_error("Invalid boundary types! (not 0, 1, 2)");
+        }
+    }
```

Comment:
This should a) not be a public member function and b) not be here at all, because if you use a proper enumeration with named constants it won't be possible to specify an invalid choice.